sharktank/sharktank/examples/paged_llm_v1.py

```diff
@@ -153,23 +203,49 @@ def prefill(self):
             attention_mask = replicate(attention_mask, tp)
             seq_block_ids_tensor = replicate(seq_block_ids_tensor, tp)
 
-        if self.dump_bins:
-            write_ndarray_to_bin(
-                token_ids.numpy(),
-                f"prefill_token_ids_{'x'.join([str(x) for x in token_ids.shape])}xi64.bin",
-            )
+        if self.dump_path is not None:
```

Comment:
Could we move the dumper work to a separate function. It would be helpful to have a simplified branch more like:

```
if self.dump_path is not None:
	self.dump("token_ids", token_ids)
	self.dump("seq_len", self.seq_len)
	...
```

We can do something similar for decode step where we embed the step number into the string. This reduces down all the replicated dump code